keps/sig-architecture/3136-beta-apis-off-by-default/README.md

```diff
@@ -0,0 +1,304 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+New beta APIs will not be enabled in clusters by default.
+Existing beta APIs and new versions of existing beta APIs, will continue to be enabled by default: 
+if v1beta.some.group is currently enabled by default and we create v1beta2.some.group, v1beta2.some.group will still be enabled by default.
+
+## Motivation
+
+Beta APIs are not considered stable and reliance upon APIs in this state leads to exposure to bugs,
+guaranteed migration pain for users when the APIs move to stable, and the risk that dependencies will
+grow around unfinished APIs.
+Enabling beta APIs by default, exacerbates these problems by making them on in nearly every cluster.
+We observed these problems as we removed long-standing beta APIs and the PRR survey tells us that over
+90% of cluster-admins leave production clusters with these APIs enabled.
+Unsuitability for production use is documented at https://kubernetes.io/docs/reference/using-api/#api-versioning 
+("The software is not recommended for production uses"), but defaulting on means they are present in nearly every 
+production cluster.
+By disabling beta APIs by default, a cluster-admin can opt-in for specific APIs without having every
+incomplete API present in the cluster.
+
+### Goals
+
+1. Disable new beta APIs by default.
+2. Continue enabling existing beta APIs and new version of existing beta APIs by default:
+   if v1beta.some.group is currently enabled by default and we create v1beta2.some.group, v1beta2.some.group will still be enabled by default.
+3. Allow enabling specific resources in beta.  Enable coolnewjobtype.v1beta1.batch.k8s.io without enabling other-neat-job.v1beta1.batch.k8s.io
+
+### Non-Goals
+
+1. Change featuregate defaults.
```

Comment:
If a KEP requires a new API _and_ a feature gate, does that mean that graduation to beta turns on the feature gate but not the API?

(which might be fine: for example, the controller for the new feature might talk to the API Server, see that the relevant API isn't served, and go to sleep).

Reply:
> if feature gates are not planned to be included here it would be nice to call it out explicitly in non-goals section
You're commenting on the line in the non-goals that says 
> ### Non-Goals
> 1. Change featuregate defaults.

Is there a way you'd like me to rephrase it to clarify?

Reply:
I recommend adding a clarifying sentence or section about what the behavior for new features might be like. The current wording is sufficient, but relies on the reader having context about what the prior situation looked like in order to model it mentally.

If this KEP is accepted, that explanatory text might eventually have a better home somewhere within https://k8s.dev/ so that contributors are clear on what's expected of their code and designs.

Reply:
added.